File: Problems/Rotate-Image/Rotate-Image.c
void rotate(int** matrix, int matrixSize, int* matrixColSize){
    for (int i = 0; i < matrixSize; i++)
        for (int j = i + 1; j < matrixSize; j++)
            swap(matrix[i] + j, matrix[j] + i);
    for (int i = 0; i < matrixSize; i++)
        for (int j = 0; j < (matrixSize >> 1); j++)
            swap(matrix[i] + j, matrix[i] + matrixSize - 1 - j);
}

void swap(int *n1, int *n2) {
    *n1 ^= *n2;
    *n2 ^= *n1;
    *n1 ^= *n2;
}

File: Problems/Rotate-Image/Rotate-Image.js
/**
 * @param {number[][]} matrix
 * @return {void} Do not return anything, modify matrix in-place instead.
 */
var rotate = function(matrix) {
    for (let i = 0; i < matrix.length; i++)
        for (let j = i + 1; j < matrix.length; j++) {
            let tmp = matrix[i][j];
            matrix[i][j] = matrix[j][i];
            matrix[j][i] = tmp;
        }
    for (let i = 0; i < matrix.length; i++)
        for (let j = 0; j < (matrix.length >> 1); j++) {
            let tmp = matrix[i][j];
            matrix[i][j] = matrix[i][matrix.length - 1 - j];
            matrix[i][matrix.length - 1 - j] = tmp;
        }
};
